### PR TITLE
Addon actions: fix slow logging

### LINKED ISSUE #2590: Action addons is slow to log events
### Issue details

A callback that logs the passed event directly to an action can incur a noticeable delay before logging.

For instance, in `cra-kitchen-sink` if you change the story to:

```js
storiesOf('Button', module)
  .add('with text', () => (
    <Button onClick={e => action('clicked')(e,e,e,e,e)}>
      {setOptions({ selectedAddonPanel: 'storybook/actions/actions-panel' })}
      Hello Button
    </Button>
```

It takes probably 2-3s for the action to be logged in the actions pane, on my iMac.

**Notes**:

1. It seems to depend on the complexity of your app exactly how slow it is. In the kitchen sink app (a CRA app) you need (say) 5 events to see it clearly. I can't make it very noticeable in the `official-storybook` app (which is raw react), even with a lot of events. In my actual app it's noticeable with a single event.

2. React <=15.6.1 makes this worse as it passes essentially 2 events by default.

3. The time is actually in the *manager* side of things; serializing the events is fairly quick.

### Please specify which version of Storybook and optionally any affected addons that you're running

- `@storybook/react@3.3.2`
- `@storybook/addon-actions@3.3.2`

### PR DESCRIPTION
**Pull request by rhalff**
All non plain objects will be serialized to a depth of 2.

Events are not treated differently anymore traversing is also cut off at the depth of 2.

All plain objects are still serialized until a depth of 10 (could probably be increased)

Properties starting with double underscore  `__` and the storybook's `STORYBOOK_` vars are not serialized and ignored.

fixes #2590

### CODE REVIEW

**Comment by Hypnosphi** [comment]
Can we make it configurable? I think there might be situation where people may need deeper nesting

**Comment by rhalff** [comment]
Yes, it possible to make it configurable.

Should I implement this by listening on the channel for options set by the `Options` addon?

**Comment by Hypnosphi** [comment]
I think exporting a configuration function from actions plugin would be more explicit

**Comment by rhalff** [comment]
Sorry, no code update yet, I was only bringing this branch up to date.

File: addons/actions/src/lib/retrocycle.js [hunk 1]
```diff
@@ -2,7 +2,6 @@ import reviver from './reviver';
 import { muteProperty } from './util';
 import { CYCLIC_KEY } from './';
 
-// eslint-disable-next-line no-control-regex
```
Comment (Hypnosphi) [inline addons/actions/src/lib/retrocycle.js]: Looks like it's still needed

Comment (rhalff) [reply addons/actions/src/lib/retrocycle.js]: Yeah, my ide reported it wasn't, will put it back.

File: addons/actions/src/index.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
+import { action, configureActions, decorateAction } from './preview';
+
 // addons, panels and events get unique names using a prefix
 export const ADDON_ID = 'storybook/actions';
 export const PANEL_ID = `${ADDON_ID}/actions-panel`;
 export const EVENT_ID = `${ADDON_ID}/action-event`;
 
-export * from './preview';
```
Comment (Hypnosphi) [inline addons/actions/src/index.js]: Yeah, unfortunately this `export *` syntax doesn't play well with `transform-runtime` https://github.com/babel/babel/issues/2877

**Comment by Hypnosphi** [comment]
Can you please add an example with deeply nested object and different `depth` configuration?

**Comment by rhalff** [comment]
It seems the performance gain still relies on setting a fixed `maxDepth` of 2 whenever an object without a name is encountered, I noticed I didn't remove that part yet: https://github.com/storybooks/storybook/pull/3133/files#diff-017c725a54046de74bba2b97b986116aR5

The decision I made there seems a bit arbitrary, but when I remove that part logging something like window will hang again. I'm not sure yet what would be a better way to prevent this.

Perhaps it's also the amount of properties/entries per object and not only the depth which causes the lag.

**Review by Hypnosphi — APPROVED** [review]

**Comment by Hypnosphi** [comment]
I assume it's ready

**Pull request merged**